aten/src/ATen/templates/StorageDerived.cpp

```diff
@@ -36,28 +57,23 @@ static int getPointerDevice(void* ptr) {
 #endif
 
 ${Storage}::${Storage}(
-  void * data, size_t size, const std::function<void(void*)> & deleter)
-  : Storage(${THStorage}_newWithDataAndAllocator(${state,}
+  void * data, 
+  size_t size, 
+  const std::function<void(void*)> & deleter) {
+  storage_impl_ = new StorageImpl(
+      ScalarType::${ScalarName},
+      size,
       InefficientStdFunctionContext::makeDataPtr(data, deleter,
 #if ${isCUDA}
       Device(kCUDA, getPointerDevice(data))
 #else
       kCPU
 #endif
-       ), size,
-     /* allocator */ nullptr
-    )) {
-    ${THStorage}_clearFlag(${state,} storage, TH_STORAGE_RESIZABLE);
-}
-
-${Storage}::~${Storage}() { }
-
-size_t ${Storage}::elementSize() const {
-  return sizeof(${ScalarType});
-}
-
-Type& ${Storage}::type() const {
-  return globalContext().getType(Backend::${Backend},ScalarType::${ScalarName});
+       ),
+     /* allocator */ nullptr,
+     StorageImpl::RESIZABLE
+    );
+  storage_impl_->clear_flag(StorageImpl::RESIZABLE);
```

Comment:
Here, clearing the resizable flag is correct, but then it is perplexing why you didn't just pass in 0 for the flag in the first case. Better yet, if you make resizable a boolean you can just say `/* resizable */ false`.